### PR TITLE
Invoice Informal Logs

### PR DESCRIPTION
**Pull request by mrie04**

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/document/impl/DocTypeBL.java [hunk 1]
```diff
@@ -202,6 +202,50 @@ public boolean isDownPayment(@NonNull final DocTypeId docTypeId)
 				&& X_C_DocType.DOCBASETYPE_APInvoice.equals(dt.getDocBaseType());
 	}
 
+	@Override
+	public boolean isFinalInvoiceOrFinalCreditMemo(@NonNull final DocTypeId docTypeId)
+	{
+		return isFinalInvoice(docTypeId) || isFinalCreditMemo(docTypeId);
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/document/impl/DocTypeBL.java:208]: u could load the doctype only once :)

File: backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java [hunk 2]
```diff
@@ -1567,6 +1573,20 @@ public final boolean isDownPayment(final org.compiere.model.I_C_Invoice invoiceR
 		return docTypeBL.isDownPayment(docTypeId);
 	}
 
+	@Override
+	public final boolean isFinalInvoiceOrFinalCreditMemo(final org.compiere.model.I_C_Invoice invoiceRecord)
+	{
+		final DocTypeId docTypeId = assumeNotNull(getDocTypeIdEffectiveOrNull(invoiceRecord), "The given C_Invoice={} needs to have a C_DocType", invoiceRecord);
```
Comment (teosarca) [inline backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java:1579]: why do we need to have a doctype?
if we don't then for sure is not a final invoice ...

Comment (mrie04) [reply backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java:1579]: Copied logic from other existing checks, but I can also return false then

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/document/impl/DocTypeBL.java (hunk 1 above)
Comment (mrie04) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/document/impl/DocTypeBL.java:208]: Did it like this so the function can be made public when this check alone is needed

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

File: backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java (hunk 2 above)
Comment (teosarca) [reply backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java:1579]: yes pls

**Pull request merged**